Preserve existing metadata labels (and other fields) in SparkApplication manifest

mohamed-mis opened the pull request:
This PR addresses an issue in the CustomObjectLauncher operator within the Kubernetes provider, where submitting a SparkApplication manifest -using the SparkKubernetesOperator- results in the loss of existing metadata labels and annotations.

The current implementation in get_body() overwrites the entire metadata object:
`self.body.metadata = {"name": self.name, "namespace": self.namespace}
`
This discards any pre-existing metadata such as labels, annotations, or custom fields.
This PR replaces the overwrite with an in-place update using dict.update().

**Testing:**
This fix has been validated with the following setup:

- Airflow Version: 3.0.6
- Deployment Method: Official Helm chart 1.18.0
- Kubernetes Provider Version: apache-airflow-providers-cncf-kubernetes==10.7.0
- Python Version: 3.12.11

SparkApplication resources submitted via the SparkKubernetesOperator now retain their original metadata labels as expected.


---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

mohamed-mis committed:
Preserve existing metadata labels in SparkApplication manifest

Ensure that CustomObjectLauncher updates only the 'name' and 'namespace' fields in the metadata without overwriting existing labels or annotations.
This change uses dict.update() to safely modify the metadata in place, preventing loss of important SparkApplication metadata during submission.

mohamed-mis committed:
misc: format committed lines for consistency

mohamed-mis committed:
Merge branch 'main' into patch-2

mohamed-mis committed:
Fix: Safely handle missing `metadata` attribute in SparkJobSpec

nevcohen reviewed (approved):
Looks good! Just add test

mohamed-mis committed:
Add unit tests for get_body method in CustomObjectLauncher

test suite for get_body logic in CustomObjectLauncher, including metadata handling

mohamed-mis committed:
Merge branch 'main' into patch-2
